### ISSUE TITLE
Idea: Make psalm aware that array shapes (with one or more non-optional fields) aren't falsey

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#756]
E.g. could limit this just to arrays where from_phpdoc is false, and possibly be less aggressive about warning in loops.

Observed: No issues emitted.
Expected: Warn that `if ($x)` is a redundant condition.

```php
<?php

function main() : void {
    $x = ['key' => 'value'];
    if ($x) {
        var_export($x);
    }
}
```

**Comment by muglug** [comment]
There’s now enough metadata on arrays to power this - I’ll add to the `more-literal-types` branch I’m working on (which is a pretty significant refactor with better support for such checks.

**Comment by muglug** [comment]
Fixed in 34a39b895cb84a03baebfdb37324ea46eae67de0
